cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -1582,6 +1583,66 @@ TEST(AsofJoinTest, BatchSequencing) {
   return TestSequencing(MakeIntegerBatches, /*num_batches=*/32, /*batch_size=*/1);
 }
 
+template <typename BatchesMaker>
+void TestSchemaResolution(BatchesMaker maker, int num_batches, int batch_size) {
+  auto l_schema =
```

Comment:
Can you add a comment explaining this test, together with a reference to the GH issue?